### PR TITLE
Add quip loader

### PR DESCRIPTION
**Pull request by shufanhao**
- **Description:** implement [quip](https://quip.com) loader
  - **Issue:** https://github.com/langchain-ai/langchain/issues/10352
  - **Dependencies:** No
  -  pass make format, make lint, make test

### CODE REVIEW

**Comment by shufanhao** [comment]
@eyurtsev can you help to take a look ?

File: libs/langchain/langchain/document_loaders/quip.py [hunk 1]
```diff
@@ -0,0 +1,224 @@
+import logging
+import re
+import xml.etree.cElementTree
+import xml.sax.saxutils
+from io import BytesIO
+from typing import List, Optional
+from xml.etree.ElementTree import ElementTree
+
+from PIL import Image
```
Comment (hwchase17) [inline libs/langchain/langchain/document_loaders/quip.py:9]: these should all be imported conditionally

**Review by hwchase17 — COMMENTED** [review]
we should also add an example notebook in the integrations documentation

**Comment by shufanhao** [comment]
> we should also add an example notebook in the integrations documentation

added.

**Comment by shufanhao** [comment]
@hwchase17 can you help to take a look ?

**Comment by shufanhao** [comment]
thank you @baskaryan

**Comment by baskaryan** [comment]
thanks @shufanhao!

**Comment by shufanhao** [comment]
@baskaryan when the new package can be released ?

**Comment by Goldabj** [comment]
Does this still work? I'm not able to fine a quip-api in the pip index?

**Comment by didn0t** [comment]
> Does this still work? I'm not able to fine a quip-api in the pip index?

I'm not sure if it ever was, there is this version https://github.com/dvwright/quipclient/tree/master.

Perhaps the langchain module should be updated to use this instead.

**Pull request merged**